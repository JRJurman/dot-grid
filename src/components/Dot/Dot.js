import { registerHtml, useUrlParams } from 'tram-one'
import './Dot.scss'

const html = registerHtml()

export default (props, children) => {
  const { size = 0.5, color = 'grey', radius = '100%' } = useUrlParams();
  const dotStyle = `
    margin: 1em;
    width: ${size}em;
    height: ${size}em;
    background: ${color};
    border-radius: ${radius};
  `
  return html`
    <div class="Dot" style=${dotStyle}>
    </div>
  `
}
