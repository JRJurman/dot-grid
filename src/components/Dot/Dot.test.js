import Dot from './Dot'

describe('Dot', () => {
  describe('with default props', () => {
    const wrapper = Dot()

    it('should match snapshot', () => {
      expect(wrapper.outerHTML).toMatchSnapshot()
    })
  })
})
