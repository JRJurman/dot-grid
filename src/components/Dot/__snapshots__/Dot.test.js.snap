// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot with default props should match snapshot 1`] = `
"<div style=\\"
    margin: 1em;
    width: 0.5em;
    height: 0.5em;
    background: grey;
    border-radius: 100%;
  \\" class=\\"Dot\\"></div>"
`;
